### DISCUSSION TITLE
Use RainbowKit without Wagmi, and just with ethers.js

### DISCUSSION THREAD

**Discussion by akorede12** [rainbow-me/rainbowkit#988] [Ideas]
So please hear me, wagmi is cool for what it's worth and the functionionality it offers, but wagmi has a couple limitations especially when it comes to calling function that return an output. So my take is this, A way to decouple rainbowkit a bit from wagmi, I understand rainbowkit is built on wagmi, but what I'm suggesting is to expose the wallet connector to ethers, a way to easily access useSigner and useProvider as just signer and provider for ethers to work with. The best of both worlds, a beautiful wallet connect modal without the limitatons of wagmi

**Comment by jxom** [comment]
> wagmi has a couple limitations especially when it comes to **calling function that return an output.**

Not sure what you mean here. Can you elaborate?

You can get an Ethers Provider & Signer using `useProvider` and `useSigner` from wagmi.

**Reply by akorede12** [reply]
Yeah I can get a signer and provider from wagmi, but I can't get it to work with ethers. I always get the error;

"Unhandled Runtime Error
Error: invalid signer or provider (argument="signerOrProvider", value="[object Object]", code=INVALID_ARGUMENT, version=contracts/5.7.0)"

Frontend code

`  const signer = useSigner();

  const [Details, setDetails] = useState([]);

  const electionTest = async () => {
    const ethersVote = new ethers.Contract(address, Electionabi.abi, signer);

    const details = await ethersVote.allElections(); // { name, creator}
    setDetails(details);
  };
`

**Reply by akorede12** [reply]
even modifying the code to use async still gets an error; 

Unhandled Runtime Error
Error: invalid signer or provider (argument="signerOrProvider", value=undefined, code=INVALID_ARGUMENT, version=contracts/5.7.0) 

`const Signer = async () => await useSigner();`

**Reply by akorede12** [reply]
So I just discovered where I got it wrong, my mistake was defining the signer the wrong way using;
`  const Signer = async () => await useSigner();`
and 
`  const signer = useSigner();` 

when it should have been 
`    const { data: signer } = useSigner()` 

it works fine now, I don't need to use wagmi hooks to interact with smart contract functions, I can just use the signer from wagmi/rainbowkit with plain ethers and it works. Thank God
